packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -108,11 +108,21 @@ abstract interface class RenderAbstractViewport extends RenderObject {
   /// when the offset of the viewport is changed by x then `target` also moves
   /// by x within the viewport.
   ///
+  /// The optional [axisDirection] is used by [RenderTwoDimensionalViewport] to
+  /// determine which [Axis] of the two should reveal. One dimensional
+  /// subclasses like [RenderViewportBase and [RenderListWheelViewport] will
+  /// assert that if an [axisDirection] is provided, that it matches the
+  /// [axisDirection] the viewport has been configured for.
```

Comment:
> So, if I need to know where to scroll my 2d scrollable to to make something visible, I have to call this twice, once for each axis?

Wouldn't this make `RevealedOffset.rect` for both individually somewhat meaningless since, when the offsets in both axises are applied (which I think is the goal - would you ever just want to only reveal in once axis?), the target will be located at neither rects? (I forget what exactly we use those rects for, I think it is relevant for a11y scrolling, see next comment?)

Reply:
> What offset will I get if I ask a 2D scroller without providing an axis?

It will assert.

Since ensureVisible must go through the scroll position, which calls getOffsetToReveal, returning two offsets is irrelevant to only one of the two scroll positions.